redisinsight/api/src/modules/triggered-functions/triggered-functions.service.ts

```diff
@@ -0,0 +1,99 @@
+import { Command } from 'ioredis';
+import { Injectable, Logger } from '@nestjs/common';
+import { concat } from 'lodash';
+import { plainToClass } from 'class-transformer';
+import { DatabaseConnectionService } from 'src/modules/database/database-connection.service';
+import { ShortLibrary, Library, Function } from 'src/modules/triggered-functions/models';
+import {
+  getLibraryInformation, getShortLibraryInformation, getLibraryFunctions,
+} from 'src/modules/triggered-functions/utils';
+import { ClientMetadata } from 'src/common/models';
+
+@Injectable()
+export class TriggeredFunctionsService {
+  private logger = new Logger('TriggeredFunctionsService');
+
+  constructor(
+    private readonly databaseConnectionService: DatabaseConnectionService,
+  ) {}
+
+  /**
+   * Get library list for particular database with name, user, totalFunctions, pendingJobs fields only
+   * @param clientMetadata
+   */
+  public async libraryList(
+    clientMetadata: ClientMetadata,
+  ): Promise<ShortLibrary[]> {
+    let client;
+    try {
+      client = await this.databaseConnectionService.createClient(clientMetadata);
+      const reply = await client.sendCommand(
+        new Command('TFUNCTION', ['LIST'], { replyEncoding: 'utf8' }),
+      );
+      const libraries = reply.map((lib: string[]) => getShortLibraryInformation(lib));
+      client.disconnect();
+      return libraries.map((lib) => plainToClass(
+        ShortLibrary,
+        lib,
+      ));
+    } catch (e) {
+      client?.disconnect();
+      this.logger.error('Unable to get database libraries', e);
+      throw e;
+    }
+  }
+
+  /**
+   * Get library details
+   * @param clientMetadata
+   * @param name
+   */
+  async details(
+    clientMetadata: ClientMetadata,
+    name: string,
+  ): Promise<Library> {
+    let client;
+    try {
+      client = await this.databaseConnectionService.createClient(clientMetadata);
```

Comment:
seriously? we are creating new client per request?

Reply:
we must use existing client here and everywhere else. For example Browser or Common (even better since triggered functions are not part of browser)

Reply:
Got it, thank you